dowhy/gcm/influence.py

```diff
@@ -0,0 +1,339 @@
+                   num_samples_conditional: int = 1000,
+                   max_num_runs: int = 5000,
+                   tolerance: float = 10 ** -4,
+                   n_jobs: int = 1,
+                   difference_estimation_func:
+                   Optional[Callable[[np.ndarray, np.ndarray], Union[np.ndarray, float]]] = None) \
+        -> Dict[Tuple[Any, Any], float]:
+    """Computes the causal strength of each edge directed to the target node.
+    The strength of an edge is quantified in terms of distance between conditional distributions of the target node in
+    the original graph and the imputed graph wherein the edge has been removed and the target node is fed a random
+    permutation of the observations of the source node. For more scientific details behind this API, please refer to
+    the research paper below.
+
+    **Research Paper**:
+    Dominik Janzing, David Balduzzi, Moritz Grosse-Wentrup, Bernhard Schölkopf. *Quantifying Causal Influences*. The
+    Annals of Statistics, Vol. 41, No. 5, 2324-2358, 2013.
+
+    :param causal_model: The probabilistic causal model for whose target node we compute the strength of incoming
+                         edges for.
+    :param target_node: The target node whose incoming edges' strength is to be computed.
+    :param parent_samples: Optional samples from the parents of the target_node. If None are given, they are generated
+                           based on the provided causal model. Providing observational data can help to mitigate
+                           misspecifications in the graph, such as missing interactions between root nodes or 
+                           confounders.
+    :param num_samples_conditional: Sample size to use for estimating the distance between distributions.
+    :param max_num_runs: The maximum number of times to resample and estimate the strength to report the average
+                         strength.
+    :param tolerance: The difference in average strength between two successive runs to terminate early without
+                      running it max_num_runs times.
+    :param n_jobs: The number of jobs to run in parallel. Set it to -1 to use all processors.
+    :param difference_estimation_func: Optional: How to measure the distance between two distributions. By default,
+                                       the difference of the variance is estimated for a continuous target node
+                                       and the KL divergence for a categorical target node.
+    :return: Causal strength of each edge.
+    """
+    if target_node not in causal_model.graph.nodes:
+        raise ValueError("Target node %s can not be found in given graph!" % target_node)
+    if is_root_node(causal_model.graph, target_node):
+        raise ValueError("Target node %s is a root node, but it requires to have ancestors!" % target_node)
+
+    sub_causal_model = ProbabilisticCausalModel(node_connected_subgraph_view(causal_model.graph, target_node))
```

Comment:
how is the sub causal model defined? Is it everything connected to the target node? good to add an inline comment.

Reply:
Yes, its the subgraph with only the upstream nodes of the target node. Added an inline comment